src/main/java/org/veupathdb/service/access/controller/ApproveEligibleStudiesController.java

```diff
@@ -0,0 +1,81 @@
+package org.veupathdb.service.access.controller;
+
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.core.Context;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.glassfish.jersey.server.ContainerRequest;
+import org.veupathdb.lib.container.jaxrs.server.annotations.Authenticated;
+import org.veupathdb.service.access.Main;
+import org.veupathdb.service.access.generated.model.EndUserPatch;
+import org.veupathdb.service.access.generated.model.EndUserPatchImpl;
+import org.veupathdb.service.access.generated.resources.ApproveEligibleAccessRequests;
+import org.veupathdb.service.access.model.ApprovalStatus;
+import org.veupathdb.service.access.model.DatasetAccessLevel;
+import org.veupathdb.service.access.model.DatasetProps;
+import org.veupathdb.service.access.model.EndUserRow;
+import org.veupathdb.service.access.model.SearchQuery;
+import org.veupathdb.service.access.service.dataset.DatasetRepo;
+import org.veupathdb.service.access.service.user.EndUserPatchService;
+import org.veupathdb.service.access.service.user.EndUserRepo;
+import org.veupathdb.service.access.util.Keys;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+
+@Authenticated
+public class ApproveEligibleStudiesController implements ApproveEligibleAccessRequests {
+  private static final long SERVICE_USER_ID = 1926010L;
+  private static final Logger LOG = LogManager.getLogger(ApproveEligibleStudiesController.class);
+
+  @Context
+  ContainerRequest _request;
+
+  @Override
+  public PostApproveEligibleAccessRequestsResponse postApproveEligibleAccessRequests(String adminAuthToken) {
+    if (!Main.config.getAdminAuthToken().equals(adminAuthToken)) {
+      LOG.info("Unauthorized user attempted to access admin endpoint.");
+      throw new NotFoundException(); // Not found, as only internal authorized user should know of endpoint existence.
+    }
+    try {
+      final List<DatasetProps> eligibleDatasets = DatasetRepo.Select.getInstance().datasetProps().stream()
+          .filter(props -> props.accessLevel == DatasetAccessLevel.PROTECTED).toList();
```

Comment:
I'll add a comment here, only protected studies are eligible for auto-approval. I thought about just letting it be controlled by presence or absence of the new property, but Cristina suggested the extra filter might be good in case there's a typo.